tests/Single/SmallSingleHTMLFormatterTest.php

```diff
@@ -35,7 +36,101 @@ public function testFormatHTMLSingleDateXsOneDay(): void
+            OfferType::event(),
+            new Status('Available', []),
+            new BookingAvailability('Available'),
+            new DateTimeImmutable($someDayInCurrentYear->format('Y-m-d') . 'T20:00:00+01:00'),
+            new DateTimeImmutable($someDayInCurrentYear->format('Y-m-d') . 'T21:30:00+01:00')
+        );
+
+        $this->assertEquals(
+            $expected,
+            $this->formatter->format($event)
+        );
+    }
+
+
+    public function testFormatHTMLSingleDateXsToday(): void
+    {
+        $event = new Offer(
+            OfferType::event(),
+            new Status('Available', []),
+            new BookingAvailability('Available'),
+            new DateTimeImmutable((new DateTimeImmutable())->format('Y-m-d') . 'T11:00:00+01:00'),
+            new DateTimeImmutable((new DateTimeImmutable())->format('Y-m-d') . 'T20:30:00+01:00')
+        );
+
+        $this->assertEquals(
+            '<span class="cf-days">Vandaag</span>',
+            $this->formatter->format($event)
+        );
+    }
+
+    public function testFormatHTMLSingleDateXsTonight(): void
+    {
+        $event = new Offer(
+            OfferType::event(),
+            new Status('Available', []),
+            new BookingAvailability('Available'),
+            new DateTimeImmutable((new DateTimeImmutable())->format('Y-m-d') . 'T18:00:00+01:00'),
+            new DateTimeImmutable((new DateTimeImmutable())->format('Y-m-d') . 'T20:30:00+01:00')
+        );
+
+        $this->assertEquals(
+            '<span class="cf-days">Vanavond</span>',
+            $this->formatter->format($event)
+        );
+    }
+
+    public function testFormatHTMLSingleDateXsTomorrow(): void
+    {
+        $tomorrow = (new DateTimeImmutable())->add(new DateInterval('P1D'));
+        $event = new Offer(
+            OfferType::event(),
+            new Status('Available', []),
+            new BookingAvailability('Available'),
+            new DateTimeImmutable($tomorrow->format('Y-m-d') . 'T18:30:00+01:00'),
+            new DateTimeImmutable($tomorrow->format('Y-m-d') . 'T22:30:00+01:00')
+        );
+
+        $this->assertEquals(
+            '<span class="cf-days">Morgen</span>',
+            $this->formatter->format($event)
+        );
+    }
+
+    public function testFormatPlainTextSingleCurrentWeek(): void
+    {
+        $offSet = (int) (new DateTimeImmutable())->format('w');
+        $daysTillSunday = 7 - $offSet;
+        $thisSunday = (new DateTimeImmutable())->add(new DateInterval('P' . $daysTillSunday . 'D'));
+        $event = new Offer(
+            OfferType::event(),
+            new Status('Available', []),
+            new BookingAvailability('Available'),
+            new DateTimeImmutable($thisSunday->format('Y-m-d') . 'T18:30:00+01:00'),
+            new DateTimeImmutable($thisSunday->format('Y-m-d') . 'T22:30:00+01:00')
+        );
+
+        $this->assertEquals(
+            '<span class="cf-meta">Deze</span> <span class="cf-days">zondag</span>',
```

Comment:
Won't this fail when running this test on a Saturday or Sunday? 🤔 If you use the Carbon library for PHP instead of the native DateTimeImmutable directly you could mock the current date & time for easier testing these relative scenario's (also in other similar tests). See https://carbon.nesbot.com/docs/#api-testing

By mocking the current date & time you wouldn't need to try to calculate a real date in the near future from the real current date & time, and you won't run into such edge cases. Because you could mock the current date & time as a specific datetime that's for example a Thursday, and then use a test date for the next Sunday (relative to that mocked Thursday).

Carbon just extends the native DateTimeImmutable, so we don't need to refactor all code in this library to use it. We can just use it in places where it's useful for easier testing, and keep using DateTimeImmutable in other places for now.